src/util/ssoUtil.js

```diff
@@ -0,0 +1,74 @@
+import { Common } from '@k8slens/extensions';
+import { logger } from './logger';
+import { extractJwtPayload } from '../renderer/auth/authUtil';
+import { AuthClient } from '../renderer/auth/clients/AuthClient';
+import * as strings from '../strings';
+
+const { Util } = Common;
+
+/**
+ *  Start authorization with MCC to get the temp access code via the
+ *  redirect URI that will use the 'lens://' protocol to redirect the user
+ *  to Lens and ultimately call back into `finishAuthorization()`.
+ * @param {Object} options
+ * @param {Object} options.config MCC Config object.
+ */
+const startAuthorization = function ({ config }) {
+  const authClient = new AuthClient({ config });
+
+  if (config.keycloakLogin) {
+    const url = authClient.getSsoAuthUrl();
+    Util.openExternal(url); // open in default browser
+  } else {
+    throw new Error(strings.ssoAuthProvider.error.basicOnly());
```

Comment:
Since this is a utility module, we don't need to use a user-facing string for this:
```suggestion
throw new Error('Cloud instance does not support SSO');
```